File: 프로그래머스/lv2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.js
function solution(elements) {
    var answer = new Set();

    elements.forEach((e, i) => {
        let acc = 0;
        for (const j in elements) {
            let idx = (parseInt(j) + i) % elements.length;
            acc += elements[idx];
            answer.add(acc);
        }
    });

    return answer.size;
}

File: 프로그래머스/lv2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cpp
#include <string>
#include <vector>
#include <set>

using namespace std;

int solution(vector<int> elements) {
    vector<int> v = elements;
    set<int> s(elements.begin(), elements.end());

    for (int i = 1; i < v.size(); ++i) // 위에서 선언할 때, 길이가 1인 연속부분수열은 계산했으니 1부터 시작
    {
        for (int j = 0; j < v.size(); ++j)
        {
            v[j] += elements[(j + i) % v.size()];
            s.insert(v[j]);
        }
    }

    return s.size();
}